### DISCUSSION TITLE
Guidelines on how to implement support of Reactor Mono and Flux into RestEasy Client

### DISCUSSION THREAD

**Discussion by soft4rchitecture** [resteasy/resteasy#3346] [Q&A]
At our company we'd need to integrate Mono and Flux types in a Java Interface and use the RESTEasy Proxy Framework to generate a client implementation using those types. 
For this to work, we'd need RESTEasy to support Reactor (Mono and Flux). 
Ref: https://docs.jboss.org/resteasy/docs/3.6.2.Final/userguide/html/Reactive.html#d4e2161
jar 'org.jboss.resteasy:resteasy-reactor' already supports the loading of FluxRxInvokerProvider and MonoRxInvokerProvider, but then the MessageBodyReaders fail and many more problems might lie ahead. 

Would any of you have any guidelines on how to do it? is there a method to add support for a new reactive types ?
Thnaks a lot

**Comment by jamezp** [comment]
What kind of failures are you seeing? I'd highly suggest using a minimum of RESTEasy 5.0.4.Final (Jakarta REST 2.1) or 6.2.1.Final (Jakarta REST 3.1). Are you running in a specific container like WildFly or JBoss EAP?

**Reply by soft4rchitecture** [reply]
I'm not running it in any container. 
i'm using RESTEasy client 6.2.1.Final
The architecture is the following: 
A REST Server implemented by SpringMVC serves a dummy data. 
A Resteasy client is used to get the data. 
One client uses Proxy Framework API + CompletionStage => Success
One client uses Programmatic API + Mono => Success
One client uses Proxy Framework API + Mono => Failure

Most of the relevant code is stored in the following gist: 
[Proxy Framework API](https://gist.github.com/soft4rchitecture/24d152aa8e26caa0ce632369c8b14e1c)

**Reply by soft4rchitecture** [reply]
Thanks a lot for your valuable time

**Reply by soft4rchitecture** [reply]
Hey, i'm writing a PR and i'll keep you updated on the status. Seems to be a bug but i'm not sure yet

**Reply by jamezp** [reply]
It looks like there is no `MessageBodyWriter` or `MessageBodyReader` for these and it expects the usage of a `RxInvoker`. I'm not too familiar with this library, however what might need to be done is maybe a new annotation like `@RxInovker(MonoRxInvoker.class)` for the target proxy.

**Reply by soft4rchitecture** [reply]
actually it’s probably a bug. I’m posting an issue and a PR along with it on tomorrow.
Roughly speaking, reactive type Publisher is exepected when the proxy framework parses the Java interface description. but Mono is written instead, so the control flow goes completely off track with invalid errors.

**Reply by jamezp** [reply]
Ah nice catch @soft4rchitecture! Much appreciated.
